admin-dev/themes/new-theme/js/components/form/form-field-toggler.ts

```diff
@@ -27,26 +27,34 @@ import {isUndefined} from '@PSTypes/typeguard';
 
 const {$} = window;
 
+// eslint-disable-next-line no-shadow
+export enum ToggleType {
+  availability = 'availability',
+  visibility = 'visibility',
+}
+
 /**
  * @param {string} disablingInputSelector - selector of input (e.g. checkbox or radio)
- *                 which on change enables/disables the element selected by targetSelector.
+ *                 which on change enables/disables or shows/hides the element selected by targetSelector.
  * @param {string} matchingValue - value which should match with disablingInput value to enable/disable related element
  * @param {string} targetSelector - selector of element which is toggled by the disablingInput.
  * @param {boolean} disableOnMatch - once disablingInput & matchingValue values match, then
- *                  if true - related element is disabled
- *                  if false - related element is enabled.
+ *                  if true - related element is disabled if toggle type is "availability" or hidden - if ToggleType is "visibility"
+ *                  if false - related element is enabled if toggle type is "availability" or visible - if ToggleType is "visibility".
+ * @param {ToggleType} toggleType - whether to toggle between enable/disable (availability) or show/hide (visibility)
  *
```

Comment:
```suggestion
 *
*  | does `disablingInput` match `matchingValue ` ? | disableOnMatch | toggleType   | target element display rule |
*  | ---------------------------------------------- | -------------- | ------------ | --------------------------- |
*  | true                                           | true           | availability | disabled                    |
*  | true                                           | false          | availability | enabled                     |
*  | true                                           | true           | visibility   | hidden                      |
*  | true                                           | false          | visibility   | visible                     |
*  | false                                          | Na.            | Na.          | Normal default behavior     |
*
```

Sometime text doesn't make things easier to understand. What do you think about adding some table ?
(Ps. You should check table content cause I am not sure about behavior I understood ^^')